concepts/05 UI Components/DataGrid/15 Columns/40 Customize Cells/2 Customize Cell Appearance.md

```diff
@@ -0,0 +1,248 @@
+To customize cell appearance, use the [columns](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/).[cellTemplate](/api-reference/_hidden/dxDataGridColumn/cellTemplate.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#cellTemplate') property or the [onCellPrepared](/api-reference/10%20UI%20Components/dxDataGrid/1%20Configuration/onCellPrepared.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onCellPrepared') function. The [columns](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/).[cellTemplate](/api-reference/_hidden/dxDataGridColumn/cellTemplate.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#cellTemplate') option allows you to replace the default cell layout with a custom markup. The [onCellPrepared](/api-reference/10%20UI%20Components/dxDataGrid/1%20Configuration/onCellPrepared.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onCellPrepared') event handler allows you to customize the existing layout after it is generated.
```

Comment:
I'm just going to comment on the order of things (structure). 

The `cssClass` property seems to be the most basic way to customize cell appearance. If I simply want to change font settings in all grid cells, that is what I would use? At least that is what I would definitely use if I need one column to stand out. No complicated code required, no conditions to be checked. I simply assign a class and that's it. So maybe this simple option should go first? 

The current introduction then mentions `cellTemplate` fist. `onCellPrepared` goes after it. But the rest of the topic reverts this order - you first give an example for `onCellPrepared`, only then for `cellTemplate`. 

Part of the difference between these two method are described in the introduction. Another part of the explanation is next to the second screenshot. Maybe it makes sense to have it all in one place. 

Then we can add some structure to the topic (section titles), like this: 

----------------------

This topic describes how you can customize cell appearance.

##  Apply a Style to All Cells in a Column 

... talk about `cssClass` ...

## Apply a Style to Individual Cells or Rows (Conditional Formatting)

### Change the Markup for the Entire Cell

Handle the onCellPrepared event... It allows you to customize the entire cell's markup after the layout has been generated... 

<example and screenshot>

### Specify a Cell Content Template

Specify a cellTemplate property. It allows you to replace cell content with custom markup, but does not give you access to the entire cell element.

<example and screenshot>